packages/gcf-utils/src/gcf-utils.ts

```diff
@@ -45,6 +47,59 @@ interface EnqueueTaskParams {
   name: string;
 }
 
+interface LogFn {
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (msg: string, ...args: any[]): void;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  (obj: object, msg?: string, ...args: any[]): void;
+}
+
+/**
+ * A logger standardized logger for Google Cloud Functions
+ */
+export interface GCFLogger {
+  [key: string]: Function;
+  trace: LogFn;
+  debug: LogFn;
+  info: LogFn;
+  warn: LogFn;
+  error: LogFn;
+  metric: LogFn;
+}
+
+let logger: GCFLogger;
+
+/**
+ * Get the singleton instance of GCFLogger
+ */
+export function getLogger(): GCFLogger {
+  if (!logger) {
+    logger = initLogger();
+  }
+  return logger;
+}
+
+export function initLogger(
+  dest?: NodeJS.WritableStream | SonicBoom
+): GCFLogger {
+  const defaultOptions: pino.LoggerOptions = {
+    customLevels: {
+      metric: 30,
+    },
+    level: 'trace',
+  };
+  const defaultDestination = pino.destination({sync: true});
+  const logger = pino(defaultOptions, dest || defaultDestination);
```

Comment:
Should we only instantiate the defaultDestination if none is provided? If one is provided, we're immediately throwing away the default one we just instantiated.

Reply:
good point. I did this from a readability perspective but didn't think of the unnecessary instantiation